lib/browser-specific.js

```diff
@@ -390,7 +397,7 @@ function walkTrees(browserTrees) {
 //                   runs. If a browser is missing, an exception will be thrown.
 //
 // Returns a map from product name to score.
-function scoreBrowserSpecificFailures(runs, expectedBrowsers) {
+function scoreBrowserSpecificFailures(runs, expectedBrowsers, options = {}) {
```

Comment:
`options` should be documented, with all recognized properties listed.

Also on a higher level, I personally would prefer passing a list of optional arguments (with default null values), in this case a filter function, instead of a catch-all `options` dict.

Reply:
Interesting; working on web-specs, we only ever use options bags and lean away from optional arguments.

The Google style guide (which again we're not necessarily using) says to use [default parameters](https://google.github.io/styleguide/jsguide.html#features-functions-parameter-return-types) 'sparingly', and to prefer [destructuring](https://google.github.io/styleguide/jsguide.html#features-objects-destructuring) instead. So this would become:

`function scoreBrowserSpecificFailures(runs, expectedBrowsers, {filterFunc = null} = {}) {`

Thoughts on that? I don't really mind, I don't have strong JS biases yet ;)

Reply:
Ahh thanks for doing the research. It makes sense as JavaScript doesn't have real named parameters and positional optional parameters are annoying to use, unlike Python. Yeah please go ahead and use the destructuring instead. At the end of the day, documentation is key!